### ISSUE TITLE
2.0.0-beta4: Win32Exception

### ISSUE THREAD

**Issue by MartinZec** [TestCentric/testcentric-gui#1035]
I installed TestCentric.GuiRunner 2.0.0-beta4 via "choco install testcentric-gui --pre" and opened it via "testcentric".
My tests are not being loaded, and I get the exception:
--Win32Exceptionsystem cannot find the file specified
  at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
  at TestCentric.Engine.Services.TestAgency.GetAgent(TestPackage package) in C:\Users\charlie\dev\TestCentric\testcentric-engine\src\TestEngine\testcentric.engine\Services\TestAgency.cs:line 153
  at TestCentric.Engine.Services.TestAgentService.GetAgent(TestPackage package) in C:\Users\charlie\dev\TestCentric\testcentric-engine\src\TestEngine\testcentric.engine\Services\TestAgentService.cs:line 81
  at TestCentric.Engine.Runners.AssemblyRunner.CreateAgentAndRunnerIfNeeded() in C:\Users\charlie\dev\TestCentric\testcentric-engine\src\TestEngine\testcentric.engine\Runners\AssemblyRunner.cs:line 236
  at TestCentric.Engine.Runners.AssemblyRunner.Explore(TestFilter filter) in C:\Users\charlie\dev\TestCentric\testcentric-engine\src\TestEngine\testcentric.engine\Runners\AssemblyRunner.cs:line 45

Is there a package missing in chocolatery?
I uninstalled the pre-release and installed 1.6.4. With 1.6.4 I was able to load my tests.

Furthermore, is possible to run NUnit4 tests with the current testcentric version?

**Comment by MartinZec** [comment]
Thank you for the information.

TestCentric is starting without any issue also without the --noload argument.
Once I load my tests I still get the same exception, in case opening TestCentric with --noload.

I also tried to delete the Nunit30Settings.xml file in AppData\Local\NUnit\ before starting TestCentric, which helped in the past to resolve some issues, but it did not help here.

**Comment by CharliePoole** [comment]
I'll see if I can replicate this myself. Any runtime or other dependencies I should focus on?

**Comment by MartinZec** [comment]
I tried with a dummy project, but this works for me. 
With my actual project, I use NUnit 3.14 and log4net 1.2. Both the dummy project and my actual project run on .NET Framework 4.8

Is there a way to log more debug output from TestCentric?

**Comment by CharliePoole** [comment]
There's currently only a 4.6.2 agent. That may be the problem and it's pretty easy for me to test. Of course, the project actually runs on whatever 4.x runtime you have installed. My tests work under 4.8. Can you post your dummy project? Could be there is some incompatibility involved.

**Comment by MartinZec** [comment]
Attached the dummy project, which was used for another issue initially. 
As mentioned, this project works fine and I can also execute tests there. I do not know why my actual project does not work with TestCentric 2.X.
[project.zip](https://github.com/TestCentric/testcentric-gui/files/13979192/project.zip)

**Comment by MartinZec** [comment]
One difference I see in the logs, which is hinting to the issue and should be reproduceable:

With my project:
ImageRequiresX86 = True --> project does not load

With dummy project:
ImageRequiresX86 = False--> project does load
Now, if I use "File -> Run as X86", also the dummy project has the same issue and cannot be loaded.

**Comment by CharliePoole** [comment]
The two cases (ImageRequiresX86 and Running as x86) are simply the difference between the engine detecting x86 and your telling to use it. NUnit console has the same thing, using an option to tell it. Since both engines only look for an X86 Test Assembly rather than at dependencies, a manual way of indicating it is needed. The general conceptwis that if you were testing an X86 SUT, you would make your test x86 as well.

Will your test  run if you retarget it at .NET 4.6.2?

**Comment by CharliePoole** [comment]
I'm now thinking that 4.6.2 vs 4.8.1 is probably not the problem. Bear in mind that you cannot load X64 and X86 code together. You can have either X64 plus AnyCPU or X86 plus AnyCPU. Is it possible you are violating this?

**Comment by CharliePoole** [comment]
As I said earlier, I do test under .NET 4.8 before releasing. The bad news is that I don't do that for X86 -  there are only uniit tests. Longer term, I'll need to add that.

**Comment by CharliePoole** [comment]
I verified this as a bug. I suspect it is present in the NuGet as well as the Chocolatey package.

**Comment by CharliePoole** [comment]
All packages fail the same way as does the current code in main. I added a simple X86 test to the package tests in both the GUI and the agent. The agent passes. The GUI fails. So the problem seems to be in the GUI itself. It may be calling the wrong agent.
